src/main/java/com/plasmideditor/rocket/web/controller/FileEditorController.java

```diff
@@ -0,0 +1,113 @@
+        } catch (SequenceValidationException e) {
+            return ResponseEntity.badRequest().body("Sequence not suitable for file: expected type is " + e.getType());
+        } catch (GenBankFileEditorException e) {
+            return ResponseEntity.badRequest().body("Fail to edit file: " + e.getMessage());
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping(path = CUT_SEQ_PATH,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> cutSequence(@RequestBody ModificationRequest request) {
+        String response;
+        try {
+            String fileContent = editService.getFileFromDB(
+                    request.getFileRequest().getFileId(),
+                    request.getFileRequest().getFileVersion()
+            );
+            Class sequenceType = editService.getSequenceType(fileContent);
+            AbstractSequence sequence = editService.cutGenBankFile(request.getSequenceInfoRequest().getStartPosition(),
+                    request.getSequenceInfoRequest().getSequence(),
+                    fileContent,
+                    sequenceType);
+            response = editService.saveSequenceToDB(request.getFileRequest().getFileId(), request.getFileRequest().getFileVersion(), sequence);
+        } catch (FileNotFoundException e) {
+            return ResponseEntity.badRequest().body("Such file doesn't exist in database");
+        } catch (UnknownSequenceType e) {
+            return ResponseEntity.badRequest().body("Sequence type is unknown");
+        } catch (GenBankFileEditorException e) {
+            return ResponseEntity.badRequest().body("Fail to edit file: " + e.getMessage());
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping(path = MODIFY_SEQ_PATH,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> modifySequence(@RequestBody ModificationRequest request) {
+        String response;
+        try {
+            String fileContent = editService.getFileFromDB(
+                    request.getFileRequest().getFileId(),
+                    request.getFileRequest().getFileVersion()
+            );
+            Class sequenceType = editService.getSequenceType(fileContent);
+            editService.validateSequence(request.getSequenceInfoRequest().getSequence(), sequenceType);
+            AbstractSequence sequence = editService.modifyGenBankFile(request.getSequenceInfoRequest().getStartPosition(),
+                    request.getSequenceInfoRequest().getSequence(),
+                    fileContent,
+                    sequenceType);
+            response = editService.saveSequenceToDB(request.getFileRequest().getFileId(), request.getFileRequest().getFileVersion(), sequence);
+        } catch (FileNotFoundException e) {
```

Comment:
https://www.baeldung.com/exception-handling-for-rest-with-spring

Reply:
done